journal/src/main/java/io/zeebe/journal/file/SparseJournalIndex.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.journal.file;
+
+import io.zeebe.journal.Journal;
+import io.zeebe.journal.JournalRecord;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+class SparseJournalIndex implements JournalIndex {
+
+  private final int density;
+  private final TreeMap<Long, Integer> indexToPosition = new TreeMap<>();
+  private final TreeMap<Long, Long> asqnToIndex = new TreeMap<>();
+  // This is added to make deleteAfter and deleteUntil easier.
+  // TODO: Check if this can be improved. https://github.com/zeebe-io/zeebe/issues/6220
+  private final TreeMap<Long, Long> indexToAsqn = new TreeMap<>();
+
+  public SparseJournalIndex(final int density) {
```

Comment:
Would be cool to have tests for this class

Reply:
There is already SparseJournalIndexTest